docs/design.md

```diff
@@ -0,0 +1,115 @@
+Design notes
+============
+
+Motivation
+----------
+
+To run GitHub Actions self-hosted runners faster and stably by making full use of
+the machine resource in the staging environment.
+
+Goals
+-----
+
+- Deploy and manage GitHub Actions self-hosted runners on multiple bare-metal
+  servers easily with Kubernetes.
+- Enable runners to finish a time-consuming initialization step before jobs are
+  assigned in order not to make users wait longer.
+- Extend lifetime of runners from outside when jobs are failed, to investigate
+  what causes the failure.
+- Notify users whether jobs are failed or not via Slack and extend the lifetime
+  from Slack.
+
+Components
+------------
+
+- Runner: `Pod` to run GitHub Actions self-hosted runner on
+- Controller manager
+  - `RunnerPool` reconciler: A controller for the `RunnerPool` custom resource(CR).
+  -  `Pod` Mutating webhook: A mutating webhook to inject a GitHub Actions
+     registration token to `Pod` `env`.
+  - Runner sweeper: A periodical task runner to sweep runners's registration
```

Comment:
I little confused by the fact that "Runner" is used in multiple means.

Reply:
Thank you. I'm gonna make the definition of the term `Runner` clearer.

Reply:
Instead, I removed the word runner for the meaning of `A periodical task runner` completely.